<?xml version="1.0"?>
<!--
  Copyright (C) 2022 LEIDOS.

  Licensed under the Apache License, Version 2.0 (the "License"); you may not
  use this file except in compliance with the License. You may obtain a copy of
  the License at

  http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
  License for the specific language governing permissions and limitations under
  the License.
-->

<!-- 
Example launch file for the CARLA Sensors Integration node
Loads parameters and configures logging for the node

 -->

 <launch>

    <arg name='role_name' default='ego_vehicle'/>

    <remap from="points_raw" to="/hardware_interface/lidar/points_raw"/>
    <remap from="image_raw" to="/hardware_interface/camera/image_raw"/>
    <remap from="image_color" to="/hardware_interface/camera/image_color"/>
    <remap from="image_rect" to="/hardware_interface/camera/image_rects"/>
    <remap from="camera_info" to="/hardware_interface/camera/camera_info"/>
    <remap from="gnss_fix_fused" to="/hardware_interface/gnss_fix_fused"/>

    <!-- data_stream_toggle -->
    <!-- Directory of vehicle config parameters -->
    <arg name="vehicle_config_dir" default="/opt/carma/vehicle/config" doc="The directory continaing vehicle config type parameters"/>
    
    <!-- Load Vehicle Characteristics -->
    <rosparam command="load" file="$(arg vehicle_config_dir)/VehicleConfigParams.yaml"/>

    <!-- Parameter Remapping -->
    <remap from="object_detection_stream" to="/object_detection_stream"/>
    <remap from="localization_stream" to="/localization_stream"/>
    <remap from="carla_lidar_stream" to="/carla_lidar_stream"/>
    <remap from="carla_camera_stream" to="/carla_camera_stream"/>
    <remap from="carla_gnss_stream" to="/carla_gnss_stream"/>
    
    <node name="carla_sensors_integration" pkg="carla_sensors_integration" type="carla_sensors_integration"/>


 </launch>